api/middleware/synchronize.js

```diff
@@ -0,0 +1,66 @@
+const logger = require('../logger')('sync middleware');
+const { cache, modelIndex } = require('./cache');
+
+const synchronizeAll = getSpecifics =>
+  cache(
+    ['sync-all', modelIndex(getSpecifics)],
+    () => async (req, res, next) => {
+      let action;
+
+      try {
+        const { modelClass, foreignKey, action: actionToReport } = getSpecifics(
+          req
+        );
+        action = actionToReport;
+
+        await modelClass.synchronize(req.body, foreignKey, false);
+
+        const synced = await modelClass.where(foreignKey).fetchAll({
+          withRelated: modelClass.withRelated
+        });
+        res.send(synced.toJSON());
+        next();
+      } catch (e) {
+        if (e.statusCode) {
+          res
+            .status(e.statusCode)
+            .send({ action, ...e.error })
+            .end();
+        } else {
+          logger.error(req, e);
+          res.status(500).end();
+        }
+      }
+    }
+  );
+
+const synchronizeSpecific = getModelToSync =>
+  cache(
+    ['sync-one', modelIndex(getModelToSync)],
```

Comment:
`getModelToSync` should probably be something else too...  maybe also `getDetails`?